### PR TITLE
Add StateVersion.Three to fix premium migration

### PR DESCRIPTION
**Pull request by eliykat**
## Type of change

- [x] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other

## Objective

Fix bug where user will not have premium after upgrading, until they log out and log back in.

Note: will be cherry-picked to `rc`

## Code changes

- `common/src/services/stateMigration.service.ts` - add a new state migration to read the premium status from the accessToken and then save it to that account
- `common/src/services/token.service.ts` - extract token decoding logic into a static method to avoid circular dependencies (ty @MGibson1)

## Before you submit

- [x] I have checked for **linting** errors (`npm run lint`) (required)
- [ ] I have added **unit tests** where it makes sense to do so (encouraged but not required)
- [ ] This change requires a **documentation update** (notify the documentation team)
- [ ] This change has particular **deployment requirements** (notify the DevOps team)

### CODE REVIEW

File: common/src/services/stateMigration.service.ts [hunk 1]
```diff
@@ -448,6 +450,23 @@ export class StateMigrationService<
     }
   }
 
+  protected async migrateStateFrom2To3(): Promise<void> {
+    const authenticatedUserIds = await this.get<string[]>(keys.authenticatedAccounts);
+    await Promise.all(
+      authenticatedUserIds.map(async (userId) => {
+        const account = await this.get<Account>(userId);
```
Comment (addisonbeck) [inline common/src/services/stateMigration.service.ts:457]: This needs to use the generic TAccount

**Review by addisonbeck — COMMENTED** [review]
Looks good, one small thing

File: common/src/services/stateMigration.service.ts [hunk 2]
```diff
@@ -448,6 +450,23 @@ export class StateMigrationService<
     }
   }
 
+  protected async migrateStateFrom2To3(): Promise<void> {
```
Comment (MGibson1) [inline common/src/services/stateMigration.service.ts:453]: I believe you're missing a call to this method from `migrate`

**Review by MGibson1 — CHANGES_REQUESTED** [review]
Looks good, but we need to use the migration 🤣

File: common/src/services/stateMigration.service.ts (hunk 2 above)
Comment (addisonbeck) [reply common/src/services/stateMigration.service.ts:453]: Good catch! StateVersion.Latest also needs to be bumped up to three

**Comment by MGibson1** [comment]
Tested with v2=>3 migration and it works

**Comment by MGibson1** [comment]
Tested with v1 => 2 => 3 and retained my premium status

File: common/src/services/stateMigration.service.ts (hunk 2 above)
Comment (eliykat) [reply common/src/services/stateMigration.service.ts:453]: Yikes! I definitely had this in another branch, obviously lost it along the way when squashing/rebasing

Comment (eliykat) [reply common/src/services/stateMigration.service.ts:456]: Should be fixed now, called from `migrate` and updated the `stateVersion` at the end of the migration method itself.

**Review by MGibson1 — APPROVED** [review]

**Review by addisonbeck — APPROVED** [review]

**Pull request merged**